packages/core/application/core-application-browser-internal/src/ui/app_router.tsx

```diff
@@ -7,7 +7,8 @@
  */
 
 import React, { FunctionComponent, useMemo } from 'react';
-import { Route, RouteComponentProps, Router, Switch } from 'react-router-dom';
+import { RouteComponentProps, Router, Switch } from 'react-router-dom';
+import { Route } from '@kbn/shared-ux-router';
```

Comment:
Core's internal rendering and routing system is supposed to remain an implementation detail. Are we certain we want to use our custom `Router` component for our AppRouter? 

Just so I understand, what will be the upsides?

Reply:
The Route from @kbn/shared-ux-router is a wrapper around the react-router-dom but other imports haven't been migrated. This PR is the component that is now shared-ux-router https://github.com/elastic/kibana/pull/131805 
One of the perks of using the @kbn/shared-ux-router is that we won't need to use useExecutionContext since the component already handles it. @majagrubic @clintandrewhall is there any context I'm missing? Thanks!

Reply:
If that's the component coming from https://github.com/elastic/kibana/pull/131805, then I *think* we don't need it for Core's internal router. But waiting for @majagrubic or @clintandrewhall's answer, in case I missed anything

Reply:
@pgayvallet if I'm following you here, you want `core` internals to continue using Route from `react-router-dom`...?  And the reason would be to reduce complexity, I assume.  I'm ok with that, so long as @lukeelmers doesn't forsee an issue with execution context being missing.

Reply:
Ah, yeah I hadn't noticed the change to core here. I don't think we need it for the internal router, as long as downstream apps are using it.

Reply:
[dc212be](https://github.com/elastic/kibana/pull/150357/commits/dc212be53f8641edcffd872e3bd355f0ae3a37ba) for changes thank you!

Reply:
>  And the reason would be to reduce complexity, I assume

Mostly to remove a risk of cyclic dependency between shared-ux component/packages and core 'internal' packages, yeah.

Thanks for making the change @rshen91